code/game/objects/machinery/robotic_cradle.dm

```diff
@@ -0,0 +1,311 @@
+#define CRADLE_NOTICE_SUCCESS 1
+#define CRADLE_NOTICE_DEATH 2
+#define CRADLE_NOTICE_NO_RECORD 3
+#define CRADLE_NOTICE_NO_POWER 4
+#define CRADLE_NOTICE_XENO_FUCKERY 5
+#define CRADLE_NOTICE_IDIOT_EJECT 6
+#define CRADLE_NOTICE_FORCE_EJECT 7
+//Cradle
+
+/obj/machinery/robotic_cradle
+	name = "Robotic Cradle"
+	icon = 'icons/obj/machines/synthpod.dmi'
+	icon_state = "idle"
+	density = TRUE
+	anchored = TRUE
+	max_integrity = 350
+	soft_armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 100, BOMB = 0, BIO = 100, FIRE = 30, ACID = 30)
+
+	var/locked = FALSE
+
+	var/repairing = FALSE
+
+	var/release_notice = TRUE
+
+	var/mob/living/carbon/human/occupant = null
+
+	var/forceeject = FALSE
+
+	var/event = 0
+
+	//It uses power
+	use_power = ACTIVE_POWER_USE
+	idle_power_usage = 15
+	active_power_usage = 10000 // It rebuilds you from nothing...
+
+	var/obj/item/radio/headset/mainship/doc/radio
+
+/obj/machinery/robotic_cradle/Initialize(mapload)
+	. = ..()
+	radio = new(src)
+
+/obj/machinery/robotic_cradle/update_icon_state()
+	if(machine_stat & NOPOWER)
+		icon_state = "No power"
+	else if(repairing)
+		icon_state = "pod_1"
+	else if (occupant)
+		icon_state = "pod_0"
+	else
+		icon_state = "idle"
+
+/obj/machinery/robotic_cradle/power_change()
+	. = ..()
+	if(is_operational() || !occupant)
+		return
+	visible_message("[src] engages the safety override, ejecting the occupant.")
+	repairing = FALSE
+	go_out(CRADLE_NOTICE_NO_POWER)
+
+/obj/machinery/robotic_cradle/process()
+	if(!occupant)
+		return
+
+	if(occupant.stat == DEAD)
+		say("Patient has expired.")
+		repairing = FALSE
+		go_out(CRADLE_NOTICE_DEATH)
+		return
```

Comment:
We kick out the dead robot, but repair_op() revives a dead robot?

Reply:
I was thinking flavor, something something robot alive to properly fix it's CPU(cloneloss)